admin-dev/themes/new-theme/js/components/form/delta-quantity-input.ts

```diff
@@ -32,48 +37,85 @@ export type DeltaQuantityConfig = {
-      modifiedQuantityClass: 'quantity-modified',
-      newQuantitySelector: '.new-quantity',
+      containerSelector: componentMap.containerSelector,
+      deltaInputSelector: componentMap.deltaInputSelector,
+      updateQuantitySelector: componentMap.updateQuantitySelector,
+      modifiedQuantityClass: componentMap.modifiedQuantityClass,
+      newQuantitySelector: componentMap.newQuantitySelector,
+      initialQuantityPreviewSelector: componentMap.initialQuantityPreviewSelector,
+      submittableInputConfig: null,
       ...config,
     };
 
     this.init();
   }
 
   private init(): void {
-    $(this.config.containerSelector).on('change', this.config.deltaInputSelector, (event: ChangeEvent) => {
-      const $deltaInput: JQuery = $(event.target);
-      const $container: JQuery = $deltaInput.closest(this.config.containerSelector);
-
-      let delta: number = parseInt(<string> $deltaInput.val(), 10);
-
-      if (Number.isNaN(delta)) {
-        delta = 0;
-      }
-      let initialQuantity = parseInt(<string> $container.data('initialQuantity'), 10);
-
-      if (Number.isNaN(initialQuantity)) {
-        initialQuantity = 0;
-      }
-      const updatedQuantity: number = initialQuantity + delta;
+    $(document).on('change', `${this.config.containerSelector} ${this.config.deltaInputSelector}`, (event: ChangeEvent) => {
+      const deltaInput: HTMLElement = event.target;
+      const $container: JQuery = $(deltaInput).closest(this.config.containerSelector);
+      const deltaQuantity = this.getDeltaQuantity(event.target);
+      const initialQuantity = this.getInitialQuantity($container);
+      const updatedQuantity: number = initialQuantity + deltaQuantity;
 
       const $newQuantity: JQuery = $container.find(this.config.newQuantitySelector);
       $newQuantity.text(updatedQuantity);
 
       const $updateElement = $container.find(this.config.updateQuantitySelector);
-      $updateElement.toggleClass(this.config.modifiedQuantityClass, delta !== 0);
+      $updateElement.toggleClass(this.config.modifiedQuantityClass, deltaQuantity !== 0);
     });
+
+    if (this.config.submittableInputConfig) {
+      new SubmittableInput(this.config.submittableInputConfig);
+      this.eventEmitter.on(ComponentsEventMap.submittableInput.submitSuccess, (input) => this.reset(input));
+    }
+  }
+
+  private reset(submittableDeltaInput: HTMLInputElement): void {
+    const $container: JQuery = $(submittableDeltaInput).closest(this.config.containerSelector);
+    const deltaQuantity = this.getDeltaQuantity(submittableDeltaInput);
+    const initialQuantity = this.getInitialQuantity($container);
+
+    $container.data('initialQuantity', initialQuantity + deltaQuantity);
+    $container.find(this.config.initialQuantityPreviewSelector).text(initialQuantity + deltaQuantity);
+    $(submittableDeltaInput).val(0);
+    $(submittableDeltaInput).data('initialValue', 0);
+    $container.find(this.config.newQuantitySelector).text(0);
```

Comment:
it did affect styling abit if i rememeber correctly. If it has more text from previous actions, then it takes more space and that space is visible after refresh. Though it was probably before i added some styling, so it should be even less important now, but I still prefer reseting it, as with big numbers it would still probably occur.